tests/test_logging.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from unittest import TestCase, mock
+
+from mbed_tools_lib.logging import (
+    log_exception,
+    set_log_level,
+    LOGGING_FORMAT,
+)
+
+
+class TestLogException(TestCase):
+    def test_log_exception_error(self):
+        mock_logger = mock.Mock(spec_set=logging.Logger)
+        mock_exception = mock.Mock(spec_set=Exception)
+
+        log_exception(mock_logger, mock_exception)
+
+        mock_logger.error.assert_called_once_with(mock_exception)
+        mock_logger.debug.assert_called_once_with(mock_exception, exc_info=True)
+
+
+@mock.patch("mbed_tools_lib.logging.logging")
+class TestSetLogLevel(TestCase):
+    def test_debug(self, mocked_logging):
```

Comment:
I'm wondering if we should look into using something like [parametrized](https://github.com/wolever/parameterized) to reduce the boilerplate test code. It claims to work with `unittest.TestCase` style tests.

Reply:
I have reduced the boilerplate a little by not having to declare the return value in each test. With that reduction it actually doesn't bother me that much to have a bit of test repetition. I find the test names quick to parse and understand whereas a parametrized test I might need to look over a couple of times to fully grok.

Reply:
Good to know that parametrize works with `unittest.TestCase` though.